Documenting NSX-T support

stuclem opened the pull request:
Some content reorg and fixes https://github.com/vmware/vic-product/issues/1088

stuclem commented:
I have updated all network-related docs to state that NSX-T logical switches can be used as well as vDS port groups. 

@yuyangbj @lgayatri @jitinkumar2018 can you please review this PR? Thanks!

yuyangbj commented on docs/user_doc/vic_vsphere_admin/bridge_network.md:
```diff
@@ -10,40 +10,44 @@ In Docker terminology, the bridge network on a VCH corresponds to the default br
-- Do not specify the same port group as the bridge network for multiple VCHs. Sharing a port group between VCHs might result in multiple container VMs being assigned the same IP address. 
-- Do not use the bridge network port group as the target for any of the other VCH networking options.
-- Do not use the bridge network for any other VM workloads.
+- You must create a dedicated vSphere port group, or an NSX Datacenter for vSphere logical switch, or an NSX-T Data Center logical switch for the bridge network for every VCH. Do not specify the same interface as the bridge network for multiple VCHs. Sharing an interface between VCHs might result in multiple container VMs being assigned the same IP address. 
```
"Do not specify the same interface" --- using interface here is strange, could you please use port group or logical switch?

stuclem replied:
Reworded to "Do not specify the same port group or logical switch as the bridge network for multiple VCHs."

yuyangbj replied:
Cool!

yuyangbj commented on docs/user_doc/vic_vsphere_admin/bridge_network.md:
```diff
@@ -10,40 +10,44 @@ In Docker terminology, the bridge network on a VCH corresponds to the default br
-- Do not use the bridge network port group as the target for any of the other VCH networking options.
-- Do not use the bridge network for any other VM workloads.
+- You must create a dedicated vSphere port group, or an NSX Datacenter for vSphere logical switch, or an NSX-T Data Center logical switch for the bridge network for every VCH. Do not specify the same interface as the bridge network for multiple VCHs. Sharing an interface between VCHs might result in multiple container VMs being assigned the same IP address. 
+- Do not use the bridge network interface as the target for any of the other VCH networking options.
```
I do not understand here, do you mean we cannot use the word "bridge"??

stuclem replied:
Reworded to "Do not use the bridge network for any of the other VCH networking options."

yuyangbj commented on docs/user_doc/vic_vsphere_admin/bridge_network.md:
```diff
@@ -10,40 +10,44 @@ In Docker terminology, the bridge network on a VCH corresponds to the default br
+<pre>--bridge-network <i>port_group_or_logical_switch_name</i></pre>
 
-If you do not specify `--bridge-network` or if you specify an invalid port group name, `vic-machine create` fails and suggests valid port groups. 
+If you do not specify `--bridge-network` or if you specify an invalid port group or logical switch name, `vic-machine create` fails and suggests valid port groups or or logical switches. 
```
two "or" here.

stuclem replied:
Fixed

yuyangbj commented on docs/user_doc/vic_vsphere_admin/bridge_network.md:
```diff
@@ -71,7 +75,7 @@ If you are using the Create Virtual Container Host wizard, stay on the Configure
 ## Example `vic-machine` Command <a id="example"></a>
 
-This example `vic-machine create` command deploys a VCH that designates an existing port group named `vch1-bridge` as the bridge network. It specifies IP addresses in the range 192.168.100.0/16 for use by user-defined bridge networks.
+This example `vic-machine create` command deploys a VCH that designates an existing interface named `vch1-bridge` as the bridge network. It specifies IP addresses in the range 192.168.100.0/16 for use by user-defined bridge networks.
```
An existing network is better?

stuclem replied:
Fixed.

yuyangbj commented on docs/user_doc/vic_vsphere_admin/container_networks.md:
```diff
@@ -15,9 +15,14 @@ Running `docker network ls` lists the container networks, and container develope
+**IMPORTANT**: 
+
+- For information about VCH networking requirements, see [Networking Requirements for VCH Deployment](network_reqs.md#vchnetworkreqs).
+- If you use NSX-T Data Center logical switches, it is not mandatory for T1 or T0 routers to be present. For container networks, if you need to access container VMs externally, you might need a T1 router to bridge overlay and underlay networks. If you only want to validate connections to container VMs via the container network, you do not require T1 and T0 routers.
```
you might need a T1..... this is just an example! So maybe we do not need to add it? We need to leave the opening to customers?

stuclem replied:
Reduced this bullet to "If you use NSX-T Data Center logical switches, you might need a T1 router to bridge overlay and underlay networks." Is this OK or should I delete the whole bullet?

yuyangbj commented on docs/user_doc/vic_vsphere_admin/interop.md:
```diff
@@ -1,67 +1,84 @@
-- Do not manually delete a VCH resource pool, the VCH endpoint VM, or container VMs. Always use the vSphere Integrated Containers plug-in for the vSphere Client or `vic-machine delete` to delete VCHs. Always use Docker commands or the vSphere Integrated Containers Management Portal to perform operations on container VMs.
-- Manually restarting container VMs can result in incorrect end-times for container operations. Do not manually restart container VMs. Always use Docker commands or the vSphere Integrated Containers Management Portal to perform operations on container VMs.
+- You can deploy VCHs to NSX Data Center for vSphere networks if those networks are configured to provide distributed port groups.
+- You can deploy VCHs that use NSX-T Data Center logical switches instead of port groups. vSphere Integrated Containers supports NSX-T Data Center version 2.3.
```
Actually, we only use the simple feature for NSXT in VIC, we can support 2.0, 2.2 and 2.3.

stuclem replied:
Fixed. What about 2.1?

yuyangbj replied:
Agree.

stuclem replied:
Added 2.1

stuclem commented on docs/user_doc/vic_vsphere_admin/bridge_network.md:
```diff
@@ -10,40 +10,44 @@ In Docker terminology, the bridge network on a VCH corresponds to the default br
+**IMPORTANT**: 
+
+- For information about VCH networking requirements, see [Networking Requirements for VCH Deployment](network_reqs.md#vchnetworkreqs).
+- If you use NSX-T Data Center logical switches, it is not mandatory for T1 or T0 routers to be present. The bridge network does not use layer 3, so does not require T1 and T0 routers.
```
These bullets are embedded under the IMPORTANT heading, so we don't need NOTE here.

stuclem commented on docs/user_doc/vic_vsphere_admin/bridge_network.md:
```diff
@@ -10,40 +10,44 @@ In Docker terminology, the bridge network on a VCH corresponds to the default br
-- Do not specify the same port group as the bridge network for multiple VCHs. Sharing a port group between VCHs might result in multiple container VMs being assigned the same IP address. 
-- Do not use the bridge network port group as the target for any of the other VCH networking options.
-- Do not use the bridge network for any other VM workloads.
+- You must create a dedicated vSphere port group, or an NSX Datacenter for vSphere logical switch, or an NSX-T Data Center logical switch for the bridge network for every VCH. Do not specify the same interface as the bridge network for multiple VCHs. Sharing an interface between VCHs might result in multiple container VMs being assigned the same IP address. 
```
The official names for NSX-V and NSX-T are now as follows:

- NSX Datacenter for vSphere (formerly NSX-V)
- NSX-T Data Center (formerly just NSX-T)

yuyangbj replied:
Thanks for reminding!

stuclem commented on docs/user_doc/vic_vsphere_admin/mgmt_network.md:
```diff
@@ -10,22 +10,23 @@ The management network is the network on which the VCH endpoint VM connects to v
 - Container VMs communicate with the VCH endpoint VM over the management network when an interactive shell is required. While the communication is encrypted, the public keys are not validated, which leaves scope for man-in-the-middle attacks. This connection is only used when the interactive console is enabled (`stdin`/`out`/`err`), and not for any other purpose. 
 - Ideally, use separate networks for the management network and container networks. 
-- You can use the same port group as the management network for multiple VCHs.
+- You can use the same interface as the management network for multiple VCHs.
```
Fixed

stuclem commented on docs/user_doc/vic_vsphere_admin/network_reqs.md:
```diff
@@ -39,26 +38,94 @@ See also [Docker container networking](https://docs.docker.com/engine/userguide/
+](https://docs.vmware.com/en/VMware-NSX-T-Data-Center/2.3/com.vmware.nsxt.admin.doc/GUID-23194F9A-416A-40EA-B9F7-346B391C3EF8.html) 
 
-The following network requirements apply to the deployment of VCHs to vCenter Server: 
+**NOTE**: If you use NSX-T Data Center logical switches, it is not mandatory for T1 or T0 routers to be present. The bridge network  does not use layer 3, so does not require T1 and T0 routers. For container networks, if you need to access container VMs externally, you might need a T1 router to bridge overlay and underlay networks. If you only want to validate connections to container VMs via the ...
```
Reworded to "For container networks, you might need a T1 router to bridge overlay and underlay networks." Remove completely or is this OK?

yuyangbj replied:
We can leave it as an example.

stuclem replied:
Updated to "For container networks, you might need a T1 router, for example to bridge overlay and underlay networks."

stuclem commented:
Hi @yuyangbj I have addressed your comments in https://github.com/vmware/vic-product/pull/2264/commits/26977cd08f05ed43096f30f3373ee4f52fa9acd5. Can you please take another look?

stuclem commented:
Removed the statement about not being able to use NSX-T/V in non-VC environments, following confirmation from @lgayatri and @jitinkumar2018 that you can indeed use them with standalone ESXi hosts. 

@yuyangbj is this OK to merge now? Thanks!

stuclem commented:
Thanks @yuyangbj !
